cmake/compiler/clang/target_warnings.cmake

```diff
@@ -93,4 +93,28 @@ macro(toolchain_cc_warning_extended)
     -Wno-tautological-compare
     )
 
-endmacro()
\ No newline at end of file
+endmacro()
+
+macro(toolchain_cc_warning_error)
+
+  # Force an error when things like SYS_INIT(foo, ...) occur with a missing header
+  zephyr_cc_option(-Werror=implicit-int)
+
+  # Prohibit date/time macros, which would make the build non-deterministic
+  # cc-option(-Werror=date-time)
```

Comment:
Don't move commented-out-code, delete it instead.

Reply:
Ok, point taken. The commented-out-code has been removed.